doc/xrcvd/transceiver-monitor-hld.md

```diff
@@ -283,6 +282,62 @@ An explicit "error clear event" can be added if some vendor's platform supports
 
 On transceiver plug in or plug out events, the port error status will be cleared.
 
+#### 1.4.3 Port Mapping Information handling ####
+
+xcvrd depends on port mapping information to update transceiver information to DB. Port mapping information contains following data:
+
+- Logical port name list. E.g. ["Ethernet0", "Ethernet4" ...]
+- Logical port name to physical port index mapping. E.g. {"Ethernet0": 1}
+- Physical port index to logical port name mapping. E.g. {1: "Ethernet0"}
+- Logical port name to ASIC ID mapping. This is useful for multi ASIC platforms.
+
+Currently, xcvrd assumes that port mapping information is never changed, so it always read static port mapping information from platform.json/port_config.ini and save it to a global data structure. However, things changed since dynamic port breakout feature. After doing a dynamic port breakout, port in CONFIG_DB could be removed or created, xcvrd cannot update transceiver information, DOM information and transceiver status information accordingly. This causes invalid data in STATE_DB. To address this issue, xcvrd should subscribe CONFIG_DB PORT table change and update port mapping information accordingly. Observer pattern will be used here to handle port mapping information change:

```

Comment:
things changed since dynamic port breakout feature **introduced**

"After doing a dynamic port breakout, port in CONFIG_DB could be removed or created"  ---> maybe can short to "ports can be added/created on the fly"

xcvrd cannot update transceiver information   -->  xcvrd cannot update transceiver information **without knowing the ports change?**

This causes invalid data in STATE_DB.  ---> data in state db not aligned with config db?